js/tree.js

```diff
@@ -372,11 +503,36 @@
 	};
 
 	$.fn.tree.defaults = {
-		dataSource: function (options, callback) {},
+		dataSource: function dataSource(options, callback) {},
 		multiSelect: false,
 		cacheItems: true,
 		folderSelect: true,
-		itemSelect: true
+		itemSelect: true,
+		/*
+		* currently calling "open" again on a folder will close it. Setting
+		* `ignoreRedundantOpens` to `true` will make the folder instead ignore
+		* the redundant call and stay open.
+		*/
+		ignoreRedundantOpens: false,
+		/*
+		* this is not necessarily indicative of how many layers deep the tree goes,
+		* only how many times `discloseAll` should be called before a warning trigger
+		* is fired for the dev to hook into. `discloseAll` will continue running
+		* after this many recursions unles the dev manually kilss it by setting
+		* `keep-disclosing` to `false`:
```

Comment:
How exactly would you do that? We could certainly bail at that point and have them listen for that bail and just restart it if they want, but, how do we decide when to bail for them? Their disclosures may be much heavier than ours. 1 layer of theirs may be worth 100 layers of ours. The decision was made to treat this more like a warning than an error. It is non-blocking, while still allowing the dev to implement some sort of mechanism to kill it, or prompt their users for whether their users would like to continue. I've included the three line example of what it would take to turn it from a non-blocking warning into a terminator. Reversing this would just mean I'd have to set the limit to infinite (meaning it wouldn't even check it) by default because we don't want people to perceive the function itself as broken ("ugh, they chose 6, but, mine could go 300 layers deep no problem") but rather the browser as broken ("ugh, when I get to layer 30 the whole browser becomes super unresponsive, how can I kill this thing at 29?"). So, either it starts throwing the event at layer 6 by default, or it never throws the event by default. In that case it's basically 6 of one half a dozen of the other...